### PR TITLE
don't append domain if uri is IP address.

### PR DESCRIPTION
**Pull request by alishir**
If user use IP address of peer, uri_complete try to append domain to the uri and call doesn't established. In this pull request first we check peer uri and if it is IP address we don't append domain to it.

### CODE REVIEW

**Comment by alishir** [comment]
The better approach to test IP in uri is using regex. I tested [some IP validation](https://www.safaribooksonline.com/library/view/regular-expressions-cookbook/9780596802837/ch07s16.html) regex with `re_regex` but the result are always non zero.

File: src/ua.c [hunk 1]
```diff
@@ -752,8 +755,15 @@ static int uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
...
-	if (0 != re_regex(uri, len, "[^@]+@[^]+", NULL, NULL)) {
+	/* Append domain if missing and uri is not IP address */
+	/* check if uri is valid IPv4 address */
+	uri_is_ip = inet_pton(AF_INET, uri, &(sa.sin_addr));
+#if HAVE_INET6
+	/* check if uri is valid IPv6 address */
+	uri_is_ip |= inet_pton(AF_INET6, uri, &(sa.sin_addr));
+#endif
+	if (0 != re_regex(uri, len, "[^@]+@[^]+", NULL, NULL) &&
+		1 != uri_is_ip) {
```
Comment (alfredh) [inline src/ua.c]: it is better to use the `sa` functions from libre, they are portable and handle
both IPv4 and IPv6.

for example:

```
	struct sa tmp;

	uri_is_ip = (0 == sa_set_str(&tmp, uri, 0));
```

File: src/ua.c [hunk 2]
```diff
@@ -739,6 +740,8 @@ static int uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
 {
 	size_t len;
 	int err = 0;
+	int uri_is_ip = 0;
```
Comment (alfredh) [inline src/ua.c]: it is probably better to use a `bool` type here since the value
is either `true` or `false`.

**Comment by alfredh** [comment]
hi @alishir 

thanks for your patch. I made a couple of comments to it :)


Alfred

**Comment by alishir** [comment]
hi @alfredh 

Thanks for you comments, I updated the PR according to your comments.

**Pull request merged**